### ISSUE TITLE
Scraping pprof profiles: specifying fewer types in helm chart causes more types to be scraped

### ISSUE THREAD

**Issue by Bruno-DaSilva** [grafana/k8s-monitoring-helm#460]
I think the profile types values parameter doesn't quite work: https://github.com/grafana/k8s-monitoring-helm/blob/feat/alloy/charts/k8s-monitoring/README.md?plain=1#L414

What happened?
I specified `profiles.pprof.types: ["cpu"]` in the values file for pprof scraping. 

Expected result
* Targets that have profiles.grafana.com/cpu.scrape="true" should be scraped for cpu
* Targets that have profiles.grafana.com/cpu.scrape="false" or null should not be scraped for cpu
* Targets that have profiles.grafana.com/memory.scrape="true" should *not* be scraped for memory
* Targets that have profiles.grafana.com/memory.scrape="false" or null should *not* be scraped for memory
* etc.

Actual result
* Targets that have profiles.grafana.com/cpu.scrape="true" are scraped for cpu
  * AND has profiles.grafana.com/memory.scrape="true" *are scraped for memory*
  * AND has profiles.grafana.com/memory.scrape="false" or null *are scraped for memory*
  * etc.
* Targets that have profiles.grafana.com/cpu.scrape="false" or null are not scraped
  * AND has profiles.grafana.com/memory.scrape="true" are not scraped for memory
  * AND has profiles.grafana.com/memory.scrape="false" or null are not scraped for memory
  * etc.

Basically, if the scrape target list (cpu=true) contains a target, it will be scraped for memory, goroutines, etc. regardless of the additional labels. I believe this is because with the type list shorter, it doesn't set enabled=false for them in the cpu pyroscope.scrape configuration. As you can see in [the docs](https://grafana.com/docs/alloy/latest/reference/components/pyroscope.scrape/#profilememory-block), enabled=true on a few of the types. So any scrape targets will default to true unless explicitly disabled.

I suggest we either update the docs on how the types values parameters work if this is expected, or change how the types param generated configs so as to scrape only the types specified and nothing more.

**Comment by petewall** [comment]
Sure enough. There are some setting that are enabled by default. I'll make the config be more explicit.

**Comment by Bruno-DaSilva** [comment]
thanks for the quick turnaround @petewall :D
